### PR TITLE
Add ArrayAnime

### PR DESCRIPTION
**Pull request by ISnackable**
ArrayAnime anime streaming site supports subbed, dubbed, chinese.

### CODE REVIEW

**Comment by ISnackable** [comment]
Hi, if code review is ok, please hold off merging as I'm thinking to change the ArrayAnime icon to transparent. Thanks.

**Comment by SuperMarcus** [comment]
> Hi, if code review is ok, please hold off merging as I'm thinking to change the ArrayAnime icon to transparent. Thanks.

Yep, no problem. I'll convert this to a draft PR, just let me know when you're ready.

**Comment by ISnackable** [comment]
Thanks, I'm just replacing the AnimeArray Site Icon png file, should be able to do that relatively quickly. 

You can review the rest of the code first, sorry if I caused any confusion, didn't really explain my previous comment well.

**Review by SuperMarcus — CHANGES_REQUESTED** [review]
Thanks for the PR and my apologies for taking so long to review it. Most of my comments are related to styling and efficiency.

* Use `.appendingPathComponent` from the `URL` class whenever possible. This helps to avoid unwrapping the optional variables;
* Use swift stdlib collection methods (`.map`, `.reduce`, etc.) whenever possible;
* Also, don't forget to add documentations to [`docs/supported-sources.md`](https://github.com/SuperMarcus/NineAnimator/blob/master/docs/supported-sources.md).

Look forward to your request for review :D

File: NineAnimator/Models/Anime Source/arrayanime.com/Arrayanime+Featured.swift [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            let seasonalAnime = try animeResponse.results
+                .map {
+                animeEntry -> AnimeLink in
+                
+                let animeURL = try URL(string: self.endpoint + "/ani/\(animeEntry.id)").tryUnwrap(.urlError)
+                
+                return AnimeLink(
+                    title: animeEntry.title,
+                    link: animeURL,
+                    image: try URL(string: animeEntry.image).tryUnwrap(.urlError),
```
Comment (ISnackable) [inline NineAnimator/Models/Anime Source/arrayanime.com/Arrayanime+Featured.swift:58]: Should the following be used here as well?
```swift
try URL( 
    protocolRelativeString: animeEntry.image,
    relativeTo: self.vercelEndpoint
).tryUnwrap() 
```

File: NineAnimator/Models/Anime Source/arrayanime.com/Arrayanime+Featured.swift [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            let recentAnime = try animeResponse.results
+                .map {
+                animeEntry -> AnimeLink in
+                
+                let animeURL = try URL(string: self.endpoint + "/ani/\(animeEntry.id)").tryUnwrap(.urlError)
+                
+                return AnimeLink(
+                    title: animeEntry.title,
+                    link: animeURL,
+                    image: try URL(string: animeEntry.image).tryUnwrap(.urlError),
```
Comment (ISnackable) [inline NineAnimator/Models/Anime Source/arrayanime.com/Arrayanime+Featured.swift:83]: Should the following be used here as well?
```swift
try URL( 
    protocolRelativeString: animeEntry.image,
    relativeTo: self.vercelEndpoint
).tryUnwrap() 
```

File: NineAnimator/Models/Anime Source/arrayanime.com/Arrayanime+Anime.swift [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                )
+                    
+                // Add each episode to every server
+                var episodeCollection = Anime.EpisodesCollection()
+                
+                let totalEpisode = Int(animeEntry.totalepisode) ?? 0
+                
+                if totalEpisode > 0 {
+                    // We assume each server contains every episode
+                    for (serverIdentifier, _) in NASourceArrayanime.knownServers {
```
[the remark this answers is not in the document]
Comment (ISnackable) [reply NineAnimator/Models/Anime Source/arrayanime.com/Arrayanime+Anime.swift:78]: Sure, optimizations ftw! 

How should I go about this? Here is my attempt but is there a better way to write this out?
```swift
var episodeCollection = Anime.EpisodeCollection()

_ = NASourceArrayanime.knownServers.map { serverIdentifier, _ in
    episodeCollection[severIdentifier] = (1...totalEpisode).map { episodeNo in
        EpisodeLink(
            // episode stuff
        )
    }
    // 
}
```

File: NineAnimator/Models/Anime Source/arrayanime.com/Arrayanime+Featured.swift (hunk 1 above)
Comment (SuperMarcus) [reply NineAnimator/Models/Anime Source/arrayanime.com/Arrayanime+Featured.swift:58]: `AnimeLink` is usually fine, as long as you're sure the website uses full URLs.

**Review by SuperMarcus — APPROVED** [review]

**Comment by uttiya10** [comment]
LGTM @SuperMarcus

**Pull request merged**